src/main/java/org/betonquest/betonquest/BetonQuest.java

```diff
@@ -343,6 +370,8 @@ private <T> T registerAndGetService(final Class<T> clazz, final T service) {
     public void onEnable() {
         instance = this;
 
+        profileProvider = new UUIDProfileProvider();
+
```

Comment:
```suggestion
        this.profileProvider = registerAndGetService(ProfileProvider.class, new UUIDProfileProvider());
```
you need this line. But call it after the other registerAndGetService methods. And your `setProfileProvider` method is unnecessary.

Reply:
Removing it. Does that mean that when other users want to register a custom profile provider they need to do it the same way `registerAndGetService` is implemented? I'm not familiar with the service manager